src/lib/rpc/svc_udp.c

```diff
@@ -272,12 +272,12 @@ static bool_t svcudp_reply(
      if (xdr_replymsg(xdrs, msg) &&
 	 (!has_args ||
 	  (SVCAUTH_WRAP(xprt->xp_auth, xdrs, xdr_results, xdr_location)))) {
-	  slen = (int)XDR_GETPOS(xdrs);
+	  slen = XDR_GETPOS(xdrs);
 	  if (sendto(xprt->xp_sock, rpc_buffer(xprt), slen, 0,
 		     (struct sockaddr *)&(xprt->xp_raddr), xprt->xp_addrlen)
 	      == slen) {
```

Comment:
The range of `size_t` is larger or equal to the `u_int` (`unsigned int`) one. So it is safe to pass the return value of `XDR_GETPOS()` to `sendto()`.

The returned type of `sendto()` is `ssize_t`, but afaik C can handle comparison of integers with different signedness safely.

Reply:
C's rules for comparing signed and unsigned values aren't safe in general, and there's a gcc warning for doing so.  If `ssize_t` is larger than `u_int` (64-bit platforms) then the comparison would be safe, but if they're the same size (32-bit platforms) then the signed value would be converted to unsigned.

Reply:
I thought that `ssize_t` was always of an higher order than `size_t` in term of integer type size, but it seems this is not the case indeed. apparently both are using the native system word size. So since `sendto()` returns a value in the range [`-1`, `INT32_MAX`] on 32-bit architectures, the only case that can lead to a bug is if `XDR_GETPOS()` returns `UINT32_MAX`.

Reply:
It's true that this comparison would work for all expected circumstances, but this code will trigger a `-Wsign-compare` warning on 32-bit platforms, so we'll need to do something else--probably assign the sendto() result to a new variable and check for `r >= 0 && (u_int)r == slen`.

Reply:
Done.